pkg/volume/csi/csi_block.go

```diff
@@ -252,22 +296,71 @@ func (m *csiBlockMapper) SetUpDevice() (string, error) {
 	}
 
 	// Call NodeStageVolume
-	stagingPath, err := m.stageVolumeForBlock(ctx, csiClient, accessMode, csiSource, attachment)
+	_, err = m.stageVolumeForBlock(ctx, csiClient, accessMode, csiSource, attachment)
 	if err != nil {
 		return "", err
 	}
 
-	// Call NodePublishVolume
-	publishPath, err := m.publishVolumeForBlock(ctx, csiClient, accessMode, csiSource, attachment, stagingPath)
+	// In SetUpDevice, NodeStageVolume is called, but NodePublishVolume isn't,
+	// therefore, there will be no device available in publishPath, yet.
+	// However, GenerateMapVolumeFunc expects SetUpDevice to return the path that the device
+	// will appear (This is needed for some existing in-tree plugins).
+	// So, publishPath is returned here, even though it is not yet available.
+	return m.getPublishPath(), nil
+}
+
+func (m *csiBlockMapper) MapPodDevice() error {
+	// Get csiSource from spec
+	if m.spec == nil {
+		klog.Error(log("blockMapper.MapPodDevice spec is nil"))
```

Comment:
comment applies throughout all code - please don't `klog` then return an `error` especially with different contexts. We should *only* return an error with all relevant context and let the top level callers log errors